enhancements/proposals/2263-multidc-client-node-tls/README.md

```diff
@@ -0,0 +1,254 @@
+client authentication - for each datacenter. These Intermediate CAs will be mirrored into their respective
+datacenters and used to sign certificates for their specific purposes, ensuring mutual trust and enabling secure TLS communication
+cluster-wide.
+
+Due to the current lack of this capability, existing users of ScyllaDBCluster are forced to use unencrypted
+client-to-node traffic in multi-datacenter deployments, or setting up the chain manually.
+
+### Goals
+
+- Propose the required API changes in `ScyllaDBCluster` and `ScyllaDBDatacenter` to support Intermediate CAs and custom trust stores.
+- Outline the chain of trust for certificate issuance.
+- Ensure that the solution enables secure, encrypted client-to-node communication across all datacenters.
+- Maintain backward compatibility for users not requiring multi-datacenter TLS.
+
+### Non Goals
+
+- do not address changes to rack-level certificates - they will continue to be managed as they are today, 
+  without splitting them into per-node certificates.
+- do not address missing TLS layer other than CQL client-node connections.
+
+## Proposal
+
+Both the `ScyllaDBCluster` and `ScyllaDBDatacenter` APIs will be extended with fields to provide user-specified references 
+to `Secrets` and `ConfigMaps` containing the Client and Serving CA TLS certificates, as well as references to custom CA bundles used as trust stores.
+This enhancement will allow users to either supply their own CA certificates for each purpose via `Secrets` or, if not provided,
+have the operator generate self-signed CAs automatically.
+
+Those Intermediate CAs together with Root CA Bundles are going to be mirrored into each datacenter.
+
+Clients connecting to any datacenter should trust the CA bundle specified in the `ScyllaDBCluster`. As a result, they
+will trust every certificate signed by each datacenter. Because CA Bundle it's going to be mirrored into each datacenter,
+clients running there will have access to it. 
+
+### User Stories
+
+#### User provided CA
+
+As a user of a multi-datacenter `ScyllaDBCluster`, I want to provide my own organization-wide CA for client-to-node
+TLS, so that all datacenters use certificates signed by the same trusted authority and clients can securely connect to
+any datacenter using a single CA bundle.
+
+#### Operator managing entire chain
+
+As a user deploying a new `ScyllaDBCluster` across multiple datacenters, I want the operator to automatically generate and
+synchronize a self-signed CA for client-to-node TLS, so that I can enable encrypted communication between clients and
+nodes in all datacenters without manual certificate management.
+
+#### Automatic certificate refresh
+
+As a user of a `ScyllaDBCluster`, I want certificates managed by the Operator to be automatically refreshed before
+their expiration, so that secure communication is maintained without manual intervention or risk of service disruption.
```

Comment:
A couple of questions regarding certificate rotation:
1. What component will be responsible for rotating serving certificates? Is it implemented by the operator, or is it offloaded to cert-manager?
2. Should we allow configuring `validFor` that's used when generating serving certificates? I think it may make sense to let a user decide how often they'd like to rotate.
3. Does Scylla support dynamic/runtime certificates loading so that if we rotate the certificate, it won't require Scylla Pod restart? If not, it may pose some risk to availability if not planned accordingly (maybe worth adding a section under Risk and Mitigation?).
4. Do we assume that CA root and intermediate certs are virtually infinitely valid (e.g., 10y) and do not need rotation?